### PR TITLE
Add support for .syn synonym files.

### LINKED ISSUE #8: use .syn files
SDCV does not use the synonyms information that is in dictionaries that come with a .syn file. In some languages synonyms are very important as a word can have many variations depending on gender, tense, etc.

### PR DESCRIPTION
**Pull request by ecraven**
Fixes #8.

### CODE REVIEW

**Comment by Dushistov** [comment]
Can you add some simple  dictionary with `.syn` somewhere to `tests/` and
some script to check that `sdcv` handle `.syn` in right way?

**Comment by ecraven** [comment]
ok, added three tests, seems to work fine, I tried with some larger local dictionaries too ;)

File: src/stardict_lib.cpp [hunk 1]
```diff
@@ -178,6 +178,18 @@ bool DictInfo::load_from_ifo_file(const std::string& ifofilename,
...
   }
 
+  p2 = strstr(p1,"\nsynwordcount=");
+  if (p2) {
+    p2+=sizeof("\nsynwordcount=")-1;
+    p3 = strchr(p2, '\n');
+    tmpstr = (gchar *)g_memdup(p2, p3-p2+1);
+    tmpstr[p3-p2] = '\0';
+    syn_wordcount = atol(tmpstr);
+    g_free(tmpstr);
```
Comment (Dushistov) [inline src/stardict_lib.cpp]: I simplify parsing of integers, so to fix build you need:
```c++
syn_wordcount = atol(std::string(p2, p3-p2);
```

File: tests/t_synonyms [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+set -e
+
+SDCV="$1"
+TEST_DIR="$2"
+
+unset SDCV_PAGER
+
+RES=$($SDCV -n --data-dir "$TEST_DIR" foo | grep result)
```
Comment (Dushistov) [inline tests/t_synonyms]: You could move repeated pattern into `sh function`, for example:

```sh
function test_word() {
    WORD="$1"
    echo "Test ${WORD}"
    RES=$($SDCV -n --data-dir "$TEST_DIR" "${WORD}" | grep result)
    if [ "result" != "$RES" ]; then
        echo "synonym for "${WORD}" should be result but was \"${RES}\""
        exit 1
    fi
}

test_word "foo"
test_word "bar"
test_word "test"
```

also to make possible code to run on machine with dictionaries with "test" word into it,
please add `-u "Test synonyms"` into `sdcv` arguments

**Comment by ecraven** [comment]
I hope I have addressed all the changes.

**Comment by Dushistov** [comment]
Thanks

**Pull request merged**